File: src/main/scala/org/rasterfun/util/SimpleFrame.scala
package org.rasterfun.util

import java.awt.Dimension
import javax.swing.{JComponent, JFrame}

/**
 * A Swing JFrame with sensible default settings.
 * Makes UI prototyping easier.
 */
class SimpleFrame(title : String, content : JComponent) extends JFrame {

  setTitle(title)
  setContentPane( content )
  setPreferredSize( new Dimension(800, 600) )

  setDefaultCloseOperation( JFrame.EXIT_ON_CLOSE )

  pack
  setVisible(true)
}

File: src/main/java/org/rasterfun/utils/SimpleFrame.java
package org.rasterfun.utils;

import javax.swing.*;
import java.awt.*;

/**
 * A Swing JFrame with sensible default settings.
 */
public class SimpleFrame extends JFrame {

    private static final int DEFAULT_WIDTH = 800;
    private static final int DEFAULT_HEIGHT = 600;

    public SimpleFrame() {
        this("", null);
    }

    public SimpleFrame(String title) {
        this(title, null);
    }

    public SimpleFrame(String title, JComponent content) {
        this(title, content, DEFAULT_WIDTH, DEFAULT_HEIGHT);
    }

    public SimpleFrame(String title, JComponent content, int initialWidth, int initialHeight) {
        setTitle(title);
        if (content != null) setContentPane(content);
        setPreferredSize(new Dimension(initialWidth, initialHeight));
        setDefaultCloseOperation(WindowConstants.EXIT_ON_CLOSE);

        pack();
        setVisible(true);
    }
}

